projects/HieraSeg/decode_head/sep_aspp_contrast_head.py

```diff
@@ -100,27 +101,32 @@ def predict_by_feat(self, seg_logits: Tensor,
         Returns:
             Tensor: Outputs segmentation logits map.
         """
-        # HieraSeg decode_head output is: (out, embedding) :tuple,
+        # HieraSeg decode_head output is: (out, embedding): tuple
         # only need 'out' here.
         if isinstance(seg_logits, tuple):
             seg_logit = seg_logits[0]
 
-        if seg_logit.size(1) == 26:
+        if seg_logit.size(1) == 26:  # For cityscapes dataset，19 + 7
+            hiera_num_classes = 7
             seg_logit[:, 0:2] += seg_logit[:, -7]
             seg_logit[:, 2:5] += seg_logit[:, -6]
             seg_logit[:, 5:8] += seg_logit[:, -5]
             seg_logit[:, 8:10] += seg_logit[:, -4]
             seg_logit[:, 10:11] += seg_logit[:, -3]
             seg_logit[:, 11:13] += seg_logit[:, -2]
             seg_logit[:, 13:19] += seg_logit[:, -1]
-        elif seg_logit.size(1) == 12:
+
+        elif seg_logit.size(1) == 12:  # For Pascal_person dataset, 7 + 5
+            hiera_num_classes = 5
             seg_logit[:, 0:1] = seg_logit[:, 0:1] + \
                 seg_logit[:, 7] + seg_logit[:, 10]
             seg_logit[:, 1:5] = seg_logit[:, 1:5] + \
                 seg_logit[:, 8] + seg_logit[:, 11]
             seg_logit[:, 5:7] = seg_logit[:, 5:7] + \
                 seg_logit[:, 9] + seg_logit[:, 11]
-        elif seg_logit.size(1) == 25:
+
+        elif seg_logit.size(1) == 25:  # For Pascal_person dataset, 7 + 5
+            hiera_num_classes = 5
```

Comment:
They are the same datasets?

Reply:
NO！I forget changing the annotation. I will fix now.😀